common/web/lm-worker/src/correction/distance-modeler.ts

```diff
@@ -537,6 +535,70 @@ namespace correction {
         maxTime = waitMillis;
       }
 
+      class ExecutionTimer {
+        private start: number;
+        private loopStart: number;
+        private maxExecutionTime: number;
+        private maxTrueTime: number;
+
+        private executionTime: number;
+
+        /**
+         * Used to track intervals in which potential context swaps by the OS may
+         * have occurred.
+         */
+        private largestIntervals: number[] = [0];
+        private iterationCount: number = 0;
+
+        constructor(maxExecutionTime: number, maxTrueTime: number) {
+          // JS measures time by the number of milliseconds since Jan 1, 1970.
+          this.loopStart = this.start = Date.now();
+          this.maxExecutionTime = maxExecutionTime;
+          this.maxTrueTime = maxTrueTime;
+        }
+
+        startLoop() {
+          this.loopStart = Date.now();
+        }
+
+        markIteration() {
+          const now = Date.now();
+          const delta = now - this.loopStart;
+          this.executionTime += delta;
+          this.iterationCount++;
+
+          if(delta) {
+            if(this.largestIntervals.length > 2 && delta > this.largestIntervals[0]) {
+              this.largestIntervals[0] = delta;
+            } else {
+              this.largestIntervals.push(delta);
+            }
+            this.largestIntervals.sort();
+          }
```

Comment:
Could you add a comment on what this is doing? It's not immediately obvious to me.